sdk/eventhub/azure-eventhubs/azure/eventhub/aio/eventprocessor/event_processor.py

```diff
@@ -205,96 +100,175 @@ def _context(self, events):
             with child:
                 yield
 
-    async def _receive(self, ownership):  # pylint: disable=too-many-statements
-        log.info("start ownership, %r", ownership)
-        partition_processor = self._partition_processor_factory()
-        partition_id = ownership["partition_id"]
-        eventhub_name = ownership["eventhub_name"]
-        consumer_group_name = ownership["consumer_group_name"]
-        owner_id = ownership["owner_id"]
-        partition_context = PartitionContext(
-            eventhub_name,
-            consumer_group_name,
-            partition_id,
-            owner_id,
-            self._partition_manager
-        )
-        partition_consumer = self._eventhub_client.create_consumer(
-            consumer_group_name,
-            partition_id,
-            EventPosition(ownership.get("offset", self._initial_event_position.value))
-        )
-
-        async def process_error(err):
-            log.warning(
-                "PartitionProcessor of EventProcessor instance %r of eventhub %r partition %r consumer group %r"
-                " has met an error. The exception is %r.",
-                owner_id, eventhub_name, partition_id, consumer_group_name, err
-            )
-            try:
-                await partition_processor.process_error(err, partition_context)
-            except Exception as err_again:  # pylint:disable=broad-except
-                log.warning(
-                    "PartitionProcessor of EventProcessor instance %r of eventhub %r partition %r consumer group %r"
-                    " has another error during running process_error(). The exception is %r.",
-                    owner_id, eventhub_name, partition_id, consumer_group_name, err_again
+    async def _receive(self, partition_id, checkpoint=None):  # pylint: disable=too-many-statements
```

Comment:
I feel like this function needs to be broken up a bit - it's quite hard to follow...

Reply:
moved process_error and close out of this function so it's shorter now.